src/Model/ModelManager.php

```diff
@@ -337,6 +337,12 @@ public function getIdentifierValues($entity)
                 continue;
             }
 
+            if (method_exists($value, '__toString')) {
```

Comment:
I'm not sure about this, `__toString` could already be used for other things, is magic, and you cannot throw inside it. How about introducing a new interface that models could implement? That would be a bit more explicit.

Reply:
new interface sounds good, not sure if people want their entities depending on sonata interface?

Reply:
Interesting, `__toString` indeed could already be used for other things. I have checked the official doc, the magic method however matches our intention - get a string representation of it

http://php.net/manual/en/language.oop5.magic.php#object.tostring

If a __toString is used for purpose other than "treated like a string", it is likely that the implementation does not follow the official interface.

What do you think?

If we can agree on the expected purpose of this magic method, it comes down to an issue where should we consider a BC breaking change.

Reply:
In my entities, I would mostly use `__toString` to return name (or something else, not the id. So, if I had a `Category` entity, my `__toString` would be returning its name. I agree with @greg0ire that `__toString` is not a good solution

Reply:
Are we talking about the same thing? it is not an `$entity->__toString()`. In this PR, the context is referring to `$entity->getId()->__toString()`

Reply:
Indeed, I overlooked this too, this looks a bit safer.